### PR TITLE
locked files needs to handle BUILD_COMPLETE better

### LINKED ISSUE #1971: problem rebuilding if case.build fails after a successful first build
I am encountering the following failure.
1) I invoke case.build successfully
2) I realize I want to change a file - and I modify a file (say in CLM)
3) I run case.build again - and there is a compilation failure
4) I fix the file so that I expect it will now compile - and I rerun case.build. I then get the following:
```
Setting build complete to False
ERROR:
ERROR env_build HAS CHANGED
  A manual clean of your obj directories is required
  You should execute the following:
    ./case.build --clean-all
```
I know that I don't want to run case.build --clean-all since everything should work except for the file that I had changed. It turns out that to fix this I need to run the following two steps:
```
rm LockedFiles/env_build.xml
./xmlchange BUILD_STATUS=0
```
This is clearly a problem that users would run into when making changes to a case.

### PR DESCRIPTION
**Pull request by jedwards4b**
If BUILD_COMPLETE is the only variable changed in env_build.xml it should not flag in check_lockedfiles

Test suite: scripts_regression_tests.py, hand testing - see test in issue #1971
Test baseline: 
Test namelist changes: 
Test status: [bit for bit, roundoff, climate changing]

Fixes #1971 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: scripts/lib/CIME/case.py [hunk 1]
```diff
@@ -1330,8 +1330,6 @@ def set_file(self, xmlfile):
         gfile = GenericXML(infile=xmlfile)
         ftype = gfile.get_id()
```
Comment (jedwards4b) [inline scripts/lib/CIME/case.py]: This is an unrelated change - It looks like a merge error, the flushalll call is at line 1328.

Comment (gold2718) [reply scripts/lib/CIME/case.py]: Gee wiz, I thought only subversion did stupid merge stuff like this. Sigh.
